### PR TITLE
Build windows 64bit binary using UCRT64

### PR DESCRIPTION
**Pull request by wader**
Is the default and recommended msystem setting. Will produce binaries that are compatible with windows 10 and later.

Also run tests for 32bit build.

Related to #2831

### CODE REVIEW

**Comment by nicowilliams** [comment]
LGTM!

**Comment by wader** [comment]
Note wait with merge, i cherry-picked https://github.com/jqlang/jq/pull/2835 ontop to try

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -196,19 +196,16 @@ jobs:
           - i386
         include:
           - arch: amd64
-            CC: x86_64-pc-msys
-            toolchain: gcc
-            msystem: msys
+            msystem: UCRT64
```
Comment (wader) [inline .github/workflows/ci.yml:199]: As we only ran windows tests for amd64 before i suspect the test failures might be some (new line character?) difference between msystem `MSYS` and `UCRT64`/`MINGW32`.

If new line issue we could use `-b` somehow but only on windows somehow? mess

**Comment by nicowilliams** [comment]
I think maybe the only thing to do is drop the question mark in one of the commits I added.

@wader don't forget not to clobber those comments, if you feel the need to push again.

**Comment by wader** [comment]
> I think maybe the only thing to do is drop the question mark in one of the commits I added.

Fixed

> @wader don't forget not to clobber those comments, if you feel the need to push again.

Not sure i follow, you mean the diff/od -c stuff? that i removed and fixedup into the first commit now

**Review by owenthereal — APPROVED** [review]

**Comment by nicowilliams** [comment]
> Not sure i follow, you mean the diff/od -c stuff? that i removed and fixedup into the first commit now

I meant "commits", but I was on my phone and the swyping came off as "comments" :)

**Comment by wader** [comment]
> I meant "commits", but I was on my phone and the swyping came off as "comments" :)

Suspected that, no worries :)

**Pull request merged**